Specify unification behavior when type variables are involved (#367)
ferrocene/specification: src/glossary.rst

^^^^^^^^^^^^^^^^

:dp:`fls_3vyodut341b5`
:dt:`Type unification` is the process by which :t:`type inference` propagates
known :t:`[type]s` across the :t:`type inference root` and assigns concrete

> :t:`[type]s` to :t:`[type variable]s`, and also a general mechanism to check

Reviewer: ```suggestion
:t:`[type]s` to :t:`[type variable]s`, as well as a general mechanism to check
```
